src/deadline/client/cli/_groups/asset_group.py

```diff
@@ -300,15 +303,65 @@ def asset_diff(root_dir: str, manifest_dir: str, raw: bool, **args):
 
 
 @cli_asset.command(name="download")
-@click.option("--farm-id", help="The AWS Deadline Cloud Farm to use.")
-@click.option("--queue-id", help="The AWS Deadline Cloud Queue to use.")
-@click.option("--job-id", help="The AWS Deadline Cloud Job to get. ")
+@click.option("--farm-id", help="The AWS Deadline Cloud Farm to use. ")
+@click.option("--queue-id", help="The AWS Deadline Cloud Queue to use. ")
+@click.option("--job-id", required=True, help="The AWS Deadline Cloud Job to get. ")
+@click.option(
+    "--manifest-out",
+    required=True,
+    help="Destination path to directory where manifest is created. ",
+)
 @_handle_error
-def asset_download(**args):
+def asset_download(job_id: str, manifest_out: str, **args):
     """
     Downloads input manifest of previously submitted job.
     """
-    click.echo("download complete")
+    if not os.path.isdir(manifest_out):
+        raise NonValidInputError(f"Specified destination directory {manifest_out} does not exist. ")
+
+    # setup config
+    config = _apply_cli_options_to_config(required_options={"farm_id", "queue_id"}, **args)
+    deadline = api.get_boto3_client("deadline", config=config)
+    queue_id = get_setting("defaults.queue_id", config=config)
+    farm_id = get_setting("defaults.farm_id", config=config)
+
+    queue = deadline.get_queue(
+        farmId=farm_id,
+        queueId=queue_id,
+    )
+
+    # assume queue role - session permissions
+    queue_role_session = api.get_queue_user_boto3_session(
+        deadline=deadline,
+        config=config,
+        farm_id=farm_id,
+        queue_id=queue_id,
+        queue_display_name=queue["displayName"],
+    )
+
+    # get input_manifest_paths from Deadline GetJob API
+    job = deadline.get_job(farmId=farm_id, queueId=queue_id, jobId=job_id)
+    attachments = job["attachments"]
+    input_manifest_paths = [manifest["inputManifestPath"] for manifest in attachments["manifests"]]
+
+    # get s3BucketName from Deadline GetQueue API
+    bucket_name = queue["jobAttachmentSettings"]["s3BucketName"]
+
+    s3_client = get_s3_client(session=queue_role_session)
+    transfer_manager = get_s3_transfer_manager(s3_client=s3_client)
+
+    # download each input_manifest_path
+    for input_manifest_path in input_manifest_paths:
+        local_file_name = Path(manifest_out, job_id + "_manifest")
+        _ = download_file_with_s3_key(
+            s3_bucket=bucket_name,
+            s3_key="DeadlineCloud/Manifests/" + input_manifest_path,
+            local_file_name=local_file_name,
+            session=queue_role_session,
+            transfer_manager=transfer_manager,
+        )
+
+    click.echo(f"Downloaded {job_id} manifest to {manifest_out}")
```

Comment:
What do you think about providing more download visibility to customer?

Reply:
I've added a the download output filename/path, as well as size. The download callback would be a nice to have, but since it's only one single manifest file, and for time's sake, I'd like to pursue other changes